test/integration-tests/strategies/deltaStrategy.ts

```diff
@@ -203,17 +207,140 @@ describe('Delta Strategy integration test', async () => {
+
+      const [collateralToAdd] = await strategy.getRequiredCollateral(strikeObj);
+      const vaultSatetBefore = await vault.vaultState();
+      const [positionBefore] = await lyraTestSystem.optionToken.getOptionPositions([positionId]);
+
+      await vault.connect(randomUser).trade(strikes[3]);
+
+      const vaultSatetAfter = await vault.vaultState();
+      expect(vaultSatetBefore.lockedAmountLeft.sub(vaultSatetAfter.lockedAmountLeft).eq(collateralToAdd)).to.be.true;
+
+      const [positionAfter] = await lyraTestSystem.optionToken.getOptionPositions([positionId]);
+      expect(positionAfter.amount.sub(positionBefore.amount).eq(defaultDeltaStrategyDetail.size)).to.be.true;
+    });
+
+    it('should be able to trade a higher strike if spot price goes up', async () => {
+      await TestSystem.marketActions.mockPrice(lyraTestSystem, toBN('3150'), 'sETH');
+
+      // triger with new strike (3550)
+      await vault.connect(randomUser).trade(strikes[4]);
+
+      // check that active strikes are updated
+      const storedStrikeId = await strategy.activeStrikeIds(1);
+      expect(storedStrikeId.eq(strikes[4])).to.be.true;
+      const positionId = await strategy.strikeToPositionId(storedStrikeId);
+      const [position] = await lyraTestSystem.optionToken.getOptionPositions([positionId]);
+
+      expect(position.amount.eq(defaultDeltaStrategyDetail.size)).to.be.true;
+    });
+    it('should revert when trying to trade the old strike', async () => {
+      await lyraEvm.fastForward(600);
+      await expect(vault.connect(randomUser).trade(strikes[3])).to.be.revertedWith('invalid strike');
+    });
+
+    const additionalDepositAmount = toBN('30');
+    it('can add more deposit during the round', async () => {
+      await vault.connect(randomUser).deposit(additionalDepositAmount);
+      const state = await vault.vaultState();
+      expect(state.totalPending.eq(additionalDepositAmount)).to.be.true;
+      const receipt = await vault.depositReceipts(randomUser.address);
+      expect(receipt.amount.eq(additionalDepositAmount)).to.be.true;
+    });
+    it('fastforward to the expiry', async () => {
+      await lyraEvm.fastForward(boardParameter.expiresIn);
+    });
+    it('should revert when closeRound is called before options are settled', async () => {
+      await expect(vault.closeRound()).to.be.revertedWith('cannot clear active position');
+    });
+    it('should be able to close closeRound after settlement', async () => {
+      await lyraTestSystem.optionMarket.settleExpiredBoard(boardId);
+
+      // settle all positions, from 1 to highest position
+      const totalPositions = (await lyraTestSystem.optionToken.nextId()).sub(1).toNumber();
+      const idsToSettle = Array.from({ length: totalPositions }, (_, i) => i + 1); // create array of [1... totalPositions]
+      await lyraTestSystem.shortCollateral.settleOptions(idsToSettle);
+      await vault.closeRound();
+
```

Comment:
should we add test that confirms that 100% of quote was exchanged to base when round ends?

Reply:
yes, will fix this in the next PR